src/main/java/teammates/storage/api/InstructorsDb.java

```diff
@@ -393,8 +393,19 @@ public boolean hasInstructorsForGoogleId(String googleId) {
         return !getInstructorsForGoogleIdQuery(googleId).keys().list().isEmpty();
     }
 
-    private Query<Instructor> getInstructorsForGoogleIdQuery(String googleId) {
-        return load().filter("googleId =", googleId);
+    private Query<Instructor> getInstructorsForGoogleIdQuery(String googleIdOrEmail) {
+        String email;
+        String googleId;
+        if (googleIdOrEmail.toLowerCase().contains("@")) {
+            email = googleIdOrEmail;
+            googleId = googleIdOrEmail.split("@")[0];
+        } else {
```

Comment:
And here it should take place only if the ID/email has no `@` in it

Reply:
I assume you're referring to the else block. My mistake previously, there will be no such cases, right?

Reply:
There will be for a short while after the release, because existing logged in users will still have their cookies in the form of without `@gmail.com`. So this condition still needs to be added.

Reply:
Fyi, the current query logic is such that if the `googleId` is a gmail, e.g. `abc@gmail.com`, then both `abc` and `abc@gmail.com` will be queried; else the `googleId` is queried as it is. Not sure if I get your concern right, but the above case will be covered as well, i.e. by the latter case.

Reply:
Unless we plan to run the script only one week after the new version is released, you need to cover this. Consider:
- Day 1, system is still at V8.18.0, someone logs in and recognized as `x` (no `@gmail.com`).
- V8.19.0 (or whichever version has this fix) is released in day 2.
- The migration script is run on day 3. `x` is now `x@gmail.com` in the DB.
- The same person logs in back on day 4, with the same login cookie that s/he gets from day 1 (remember that cookies are good for 7 days). S/he will not get the entities with `x` as the ID as they have been migrated.

While it's not a big deal (s/he can just logout and re-login), it will be good if we can cover this even if temporarily.

Reply:
I see, I overlooked the period between after migration and merging of new code.